addons/mail/static/src/models/attachment.js

```diff
@@ -280,6 +302,17 @@ registerModel({
             }
             return;
         },
+        async _onChangeIsEmpty() {
+            if (this.isEmpty) {
+                const response = await this.env.services.rpc({
+                    route: '/mail/update_link_preview',
+                    params: {
+                        attachment_id: this.id,
+                    }
+                });
+                this.messaging.models['Attachment'].insert(response);
```

Comment:
I think it would be easier to just make an `RPC` and handle response using longpolling.
This approach scales much better with time.